src/modules/keyboard.coffee

```diff
@@ -46,18 +46,23 @@ class Keyboard
   _initDeletes: ->
     this.addHotkey([dom.KEYS.DELETE, dom.KEYS.BACKSPACE], (range, hotkey) =>
       if range? and @quill.getLength() > 0
-        if range.start != range.end
-          @quill.deleteText(range.start, range.end, Quill.sources.USER)
+        { start, end } = range
+        if start != end
+          # if the surrounding characters are both spaces,
+          # kill the preceding space to prevent leaving double-spaces
+          if ' ' == @quill.getText(start - 1, start) == @quill.getText(end, end + 1)
```

Comment:
Can a check be added to make sure `start - 1 >= 0` and `end + 1 < @quill.getLength()`?